### DISCUSSION TITLE
Zephyr RTOS on ESP32 — First Steps · Espressif Developer Portal

### DISCUSSION THREAD

**Discussion by giscus** [espressif/developer-portal#365] [General]
# Zephyr RTOS on ESP32 — First Steps · Espressif Developer Portal

The developer resources in just one place!

https://developer.espressif.com/blog/zephyr-rtos-on-esp32-first-steps/

**Comment by samuelhrqe** [comment]
Hello! My name is Samuel Henrique, I’m from Brazil, and I’m trying to learn about Zephyr RTOS on the ESP-WROOM-32 board with 38 pins. I attempted to run the `samples/hello_world`, built it, and flashed it, but when checking the logs with `west espressif monitor`, I noticed the board is constantly restarting, showing the following log:
```I (49) soc_init: ESP Simple boot
I (49) soc_init: compile time Dec 16 2024 23:06:26
W (49) soc_init: Unicore bootloader
I (50) spi_flash: detected chip: generic
I (54) spi_flash: flash io: dio
I (57) soc_init: chip revision: v1.0
E (60) soc_init: You are using ESP32 chip revision (1) that is unsupported. While it may work, it could cause unexpected behavior or issues.
E (73) soc_init: Proceeding with this ESP32 chip revision is not recommended unless you fully understand the potential risk and limitations.
E (85) soc_init: If you choose to continue, please enable the 'CONFIG_ESP32_USE_UNSUPPORTED_REVISION' in your project configuration.
E (97) boot: HW init failed, aborting
Fatal exception (0): IllegalInstruction
epc1=0x400d13b2, epc2=0x00000000, epc3=0x00000000, excvaddr=0x00000000, depc=0x00000000
0x400d13b2: abort at /home/samuel/Projetos/zephyr-rtos/zephyr/lib/libc/common/source/stdlib/abort.c:11
```

I don’t know how to fix this.

I installed Zephyr by following the guide:  
[https://docs.zephyrproject.org/latest/develop/getting_started/index.html](https://docs.zephyrproject.org/latest/develop/getting_started/index.html)

I tried running the example using this guide:  
[https://docs.zephyrproject.org/latest/boards/espressif/esp32_devkitc_wroom/doc/index.html](https://docs.zephyrproject.org/latest/boards/espressif/esp32_devkitc_wroom/doc/index.html)

Both results were the same as shown above. Could you please help me?

I'm using the `zephyr-sdk-0.17.0` version.

**Comment by wmrsouza** [comment]
Hi Samuel, thaks for the contact!
You are receiving this `warning`message and probably facing this chash because the ESP32 `Rev 1` is not supported on Zephyr. Only  ESP32 with chip revisions greater than or equal to `Rev 3` are support on Zephyr. If you, despite of the `warning message`, choose enable `CONFIG_ESP32_USE_UNSUPPORTED_REVISION` by adding `-DCONFIG_ESP32_USE_UNSUPPORTED_REVISION=y` to the end of your `west build` command be aware you can face `unexpected behavior` while executing your Zephyr application. Enabling the`CONFIG_ESP32_USE_UNSUPPORTED_REVISION` only inhibits the `warning message`. To solve the crash you are facing the first recommended measure is continue you tests and studies with a board containing an ESP32 with minimum `Rev 3` chip revision.

**Reply by samuelhrqe** [reply]
Hi Márcio Ribeiro. Thank you for your reply. 
After conducting some research and carefully reviewing the log output, I noticed the `CONFIG_ESP32_USE_UNSUPPORTED_REVISION=y` flag. I added it first to the `prj.conf` file and then to the `board/esp32_devkitc_wroom.conf` file. It worked well! This resolved my issue, the warning message and the crash are now gone.

Regarding the chip revision, I was caught by surprise, I had never noticed this while developing with ESP-IDF and FreeRTOS. Although I managed to fix it, I think I’ll need to purchase an ESP32 with a newer revision to avoid potential problems in the future. Thank you very much!

**Comment by kavandhamsania** [comment]
Hello Espressif Community,

I am currently working on a project involving the ESP32-S3 and Zephyr RTOS, and I'm trying to configure the ESP32 as an I2C slave Mode. However, I'm running into issues getting the slave mode to function correctly.

I've set up the device tree and enabled the necessary Zephyr I2C configs, but it appears that the I2C slave mode is not working as expected. From my investigation, it seems that the current i2c_esp32.c driver may not support slave functionality fully, or may require patching.

Any guidance, sample code, or suggestions would be greatly appreciated.

Thank you for your support!

Best regards,
Kavan
